### PR TITLE
Update react to bootstrap 5.

### PR DESCRIPTION
**Pull request by mshima**
Depends on https://github.com/jhipster/react-jhipster/pull/75.

Tasks:
- [x] run side-by-side and look for unwanted changes.
- [x] check rtl support.
- [x] menu translation not applied
- [x] login close button
- [x] pagination alignment
- [ ] switch react-jhipster to release.

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/react/package.json [hunk 1]
```diff
@@ -5,21 +5,21 @@
...
-    "bootstrap": "4.6.1",
+    "bootstrap": "5.1.3",
     "bootswatch": "5.1.3",
     "lodash": "4.17.21",
     "path-browserify": "1.0.1",
     "react": "17.0.2",
     "react-dom": "17.0.2",
     "react-hook-form": "7.19.1",
-    "react-jhipster": "0.17.0",
+    "react-jhipster": "github:jhipster/react-jhipster#main",
```
Comment (deepu105) [inline generators/client/templates/react/package.json:15]: This is temporary right?

File: generators/client/files-common.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ const commonFiles = {
       ],
     },
     {
-      condition: generator => generator.enableI18nRTL,
+      condition: generator => generator.enableI18nRTL && !generator.clientFrameworkReact,
```
Comment (deepu105) [inline generators/client/files-common.js:65]: Why do we skip this? Is it all handled by the post-rtlcss plugin now?

File: generators/client/templates/react/package.json (hunk 1 above)
Comment (deepu105) [inline generators/client/templates/react/package.json:15]: This is temporary right?

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs [hunk 3]
```diff
@@ -108,7 +108,7 @@ export const UserManagement = (props: RouteComponentProps<any>) => {
       <h2 id="user-management-page-heading" data-cy="userManagementPageHeading">
         <Translate contentKey="userManagement.home.title">Users</Translate>
         <div className="d-flex justify-content-end">
-          <Button className="mr-2" color="info" onClick={handleSyncList} disabled={loading}>
+          <Button className="me-2" color="info" onClick={handleSyncList} disabled={loading}>
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:111]: Is 'me' a new class in Bootstrap? I assume margin-end? Should we update other margin classes as well?

**Review by deepu105 — APPROVED** [review]
Look good. Great work Marcelo. Do update the repo to point to npm release before merge

**Comment by deepu105** [comment]
Is there ticket with bounty for this?

File: generators/client/files-common.js (hunk 2 above)
Comment (mshima) [reply generators/client/files-common.js:65]: Yes.
bootstrap 5 have native rtl support using rtlcss.
All in one css is not officially supported, but looks to work great.

File: generators/client/templates/react/package.json (hunk 1 above)
Comment (mshima) [reply generators/client/templates/react/package.json:15]: Sure, once a new version is released, we switch to it.

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs (hunk 3 above)
Comment (mshima) [reply generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:111]: As far as I know I've updated everything.

Comment (mshima) [reply generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:111]: Most of the changes are on react-jhipster side.

**Comment by deepu105** [comment]
Awesome. Let me know if there is a bounty for this. I'll assign here if not

**Comment by mshima** [comment]
@deepu105 there isn't.

**Comment by deepu105** [comment]
Done

**Comment by mshima** [comment]
> Look good. Great work Marcelo. Do update the repo to point to npm release before merge

It just missing react-jhipster release now.
@pascalgrimaud @deepu105 whenever you have time 😄. Thanks.

**Comment by pascalgrimaud** [comment]
Ok I'll try to not forget tomorrow :-D

**Comment by deepu105** [comment]
I will do it tomorrow unless Pascal beats me to it

**Comment by deepu105** [comment]
@mshima @pascalgrimaud v0.18 released

**Comment by pascalgrimaud** [comment]
@deepu105 : :1st_place_medal:  you beat me because I forgot this morning. I need more :coffee: 
Thanks mate ;)

**Comment by mshima** [comment]
@pascalgrimaud bounty claimed https://opencollective.com/generator-jhipster/expenses/59117.

**Comment by pascalgrimaud** [comment]
@mshima : approved

**Pull request merged**